feature_engine/prediction/base_predictor.py

```diff
@@ -0,0 +1,235 @@
+    numerical, the values are first sorted into bins of equal-width or equal-frequency.
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of input variables. If None, the estimator will use all variables as
+        input features (except datetime).
+
+    bins: int, default=5
+        If the dataset contains numerical variables, the number of bins into which
+        the values will be sorted.
+
+    strategy: str, default='equal_width'
+        Whether the bins should of equal width ('equal_width') or equal frequency
+        ('equal_frequency').
+
+    Attributes
+    ----------
+    variables_categorical_:
+        The group of categorical input variables that will be used for prediction.
+
+    variables_numerical_:
+        The group of numerical input variables that will be used for prediction.
+
+    pipeline_:
+        A Sickit-learn Pipeline with a dicretiser and/or encoder. Used to determine the
+        mean target value per category or bin, per variable.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Learn the mean target value per category or per bin, for each variable.
+    predict:
+        Predict using the average of the target mean value across variables.
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+    Adapted from:
+
+    .. [1] Miller, et al. "Predicting customer behaviour: The University of Melbourne’s
+        KDD Cup report". JMLR Workshop and Conference Proceeding. KDD 2009
+        http://proceedings.mlr.press/v7/miller09/miller09.pdf
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        bins: int = 5,
+        strategy: str = "equal_width",
+    ):
+        # boolean value can be interpreted as an integer
+        if not isinstance(bins, int) or isinstance(bins, bool):
+            raise ValueError(f"bins must be an integer. Got {bins} instead.")
+
+        if strategy not in ["equal_width", "equal_frequency"]:
+            raise ValueError(
+                "strategy takes only values equal_width or equal_frequency. Got "
+                f"{strategy} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.bins = bins
+        self.strategy = strategy
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Learn the mean target value per category or bin.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : pandas series of shape = [n_samples,]
+            The target variable.
+        """
+        # check if 'X' is a dataframe
+        _is_dataframe(X)
+
```

Comment:
Hi @Morgan-Sell 

would you be so kind to experiment with adding this check and see if the tests still pass:

```
# Check that X and y have correct shape
X, y = check_X_y(X, y, dtype=None)
```

check_X_y is from sklearn and supposedly checks that X and y fulfill some criteria.

Reply:
I played w/ `check_X_y`. The function returns numpy arrays which lack a dtype and cause errors downstream.

I tried transforming `X` and `y` into pandas dataframe and series after `check_X_y` was executed. This caused ` ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().` to occur for various unit tests.

I think the root of both errors exists when the MeanEncoder is applied.